drivers/ieee802154/ieee802154_b91.c

```diff
@@ -613,14 +613,11 @@ static struct ieee802154_radio_api b91_radio_api = {
 
 /* IEEE802154 driver registration */
 #if defined(CONFIG_NET_L2_IEEE802154) || defined(CONFIG_NET_L2_OPENTHREAD)
-NET_DEVICE_INIT(b91_154_radio, CONFIG_IEEE802154_B91_DRV_NAME,
-		b91_init, NULL, &data, NULL,
-		CONFIG_IEEE802154_B91_INIT_PRIO,
-		&b91_radio_api, L2,
-		L2_CTX_TYPE, MTU);
+NET_DEVICE_DT_INST_DEFINE(0, b91_init, NULL, &data, NULL,
```

Comment:
same as https://github.com/zephyrproject-rtos/zephyr/pull/47805/files#r921580120. I'll give you some time to think about it before reviewing the rest of these